samples/nrf9160/modem_shell/README.rst

```diff
@@ -0,0 +1,380 @@
+
+Examples
+--------
+
+HTTP download::
+
+   curl http://curl.server.url/download_500k_file.zip
+
+Socket tool
+===========
+
+MoSh command: "sock"
+
+Socket tool is used to create and manage socket connections as well as send and receive data.
+
+Examples
+--------
+
+Open and connect to an IP address and port (IPv4 TCP socket)::
+
+   sock connect -a 111.222.111.222 -p 20000
+
+Open and connect to hostname and port (IPv4 TCP socket)::
+
+   sock connect -a google.com -p 20000
+
+Open and connect IPv6 TCP socket and bind to a port::
+
+   sock connect -a 1a2b:1a2b:1a2b:1a2b::1 -p 20000 -f inet6 -t stream -b 40000
+
+Open IPv6 UDP socket::
+
+   sock connect -a 1a2b:1a2b:1a2b:1a2b::1 -p 20000 -f inet6 -t dgram
+
+Open raw socket::
+
+   sock connect -f packet -t raw
+
+Open a socket to non-default PDP context::
+
+   ltelc connect -a nondefault.context.com
+   sock connect -f packet -t raw -I 1
+
+Send string through socket::
+
+   sock send -i 0 -d testing
+
+Send 100kB of data and show throughput statistics::
+
+   sock send -i 0 -l 100000
+
+Send data periodically with 10s interval::
+
+   sock send -i 0 -e 10 -d test_periodic
+
+Calculate receive throughput::
+
+   <do whatever is needed to make device receive data after some time>
+   sock recv -i 0 -r -l 1000000
+   sock recv -i 0
+   sock recv -i 0
+
+Close socket::
+
+   sock close -i 0
+
+Use RAI settings::
+
+   ltelc funmode -4
+   sock rai --rai_enable
+   ltelc funmode -1
+   sock connect -a 111.222.111.222 -p 20000
+   sock rai -i 0 --rai_last
+   sock send -i 0 -d testing
+
+List open sockets::
+
+   sock list
+
+SMS tool
+========
+
+MoSh command: "sms"
+
+SMS tool can be used for sending and receiving SMS messages.
+
+Examples
+--------
+
+Register SMS service so messages can be received if SIM subscription supports it:
+
+   sms reg
+
+Send SMS message (registration is done automatically if not already done)::
+
+   sms send -n +987654321 -m testing
+
+GNSS
+====
+
+MoSh command: "gnss"
+
+GNSS provides commands for searching the location of the device.
+
+Examples
+--------
+
+Start GNSS tracking and then stop it::
+
+   gnss start
+   gnss stop
+
+Disable LTE, enable all GNSS output and start continuous tracking with power saving enabled::
+
+   ltelc funmode --lteoff
+   gnss output 2 1 1
+   gnss mode cont
+   gnss config powersave perf
+   gnss start
+
+Enable LTE PSM, only NMEA output, automatic A-GPS data fetching and start periodic fixes with 5 minute interval and 120 second timeout::
+
+   ltelc psm -e
+   gnss output 0 1 0
+   gnss agps automatic enable
+   gnss mode periodic 300 120
+   gnss start
+
+Enable Low Noise Amplifier (LNA) for use with the on-board GPS antenna::
+
+   gnss lna enable
+
+FOTA
+====
+
+MoSh command: "fota"
+
+FOTA can be used to perform software updates over-the-air. However, you need to know what updates are available in NRF servers in order to use this feature completely.
```

Comment:
What should we document here? Fota feature is made to use our servers that are publicly available. However, you need to know the names of the files that you want to update to, and hence this is useful for some stakeholders.
Any comments @rlubos

Reply:
> Fota feature is made to use our servers that are publicly available.

Is that related to the modem FOTA or the application FOTA as well (i. e. do/will we provide a snapshot images of the MoSH sample)? Do we maintain a list of available modem update images somewhere in the docs already?

Reply:
It's possible to do both modem and application FOTA with this command. Currently we have no plans to maintain a public list of available images on our test servers. However, this feature is used by selected users and customers to whom we will communicate the image names they can use.

We could create a JIRA item to make this more generic later, so that the user could also specify the address of the server. This usually also involves certificates, so it's not just a matter of allowing the user to provide the address.

Reply:
I've updated the description slightly to mention that this is intended for selected users. @rlubos, @tokangas  please check if this sounds fine.

Reply:
Looks good to me.